lib/fog/aws/models/storage/directories.rb

```diff
@@ -11,6 +11,7 @@ def all
           load(data)
         end
 
+        # **Warning!** `s3.directories.get` retrieves and caches meta data for the first 10,000 objects in the bucket, which can be very expensive. When possible use `s3.directories.new`.
```

Comment:
Not 100% sure this will work as expected.
Got my inspiration from https://github.com/orrin-naylor-instacart/fog-aws/blob/orrin/directories_get-warning/lib/fog/aws/models/storage/directory.rb#L60
https://www.rubydoc.info/github/fog/fog-aws/Fog%2FAWS%2FStorage%2FDirectory:location=

LMK if this isn't how this works